File: src/main/java/no/nav/sosialhjelp/soknad/business/db/repositories/soknadunderarbeid/BatchSoknadUnderArbeidRepository.java
package no.nav.sosialhjelp.soknad.business.db.repositories.soknadunderarbeid;


import no.nav.sosialhjelp.soknad.domain.SoknadUnderArbeid;

import java.util.List;
import java.util.Optional;

public interface BatchSoknadUnderArbeidRepository {

    Optional<Long> hentSoknadUnderArbeidIdFromBehandlingsIdOptional(String behandlingsId);
    List<Long> hentGamleSoknadUnderArbeidForBatch();
    void slettSoknad(Long soknadUnderArbeidId);
    List<SoknadUnderArbeid> hentForeldedeEttersendelser();

}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/db/repositories/soknadunderarbeid/BatchSoknadUnderArbeidRepository.kt
package no.nav.sosialhjelp.soknad.db.repositories.soknadunderarbeid

interface BatchSoknadUnderArbeidRepository {
    fun hentSoknadUnderArbeid(behandlingsId: String): SoknadUnderArbeid?

    fun hentSoknadUnderArbeid(soknadUnderArbeidId: Long): SoknadUnderArbeid?

    fun hentGamleSoknadUnderArbeidForBatch(): List<Long>

    fun slettSoknad(soknadUnderArbeidId: Long?)

    fun hentForeldedeEttersendelser(): List<SoknadUnderArbeid>
}
